File: peek_plugin_docdb/tuples/DocumentTuple.py
from typing import Dict

from peek_plugin_docdb._private.PluginNames import docDbTuplePrefix
from vortex.Tuple import Tuple, addTupleType, TupleField


@addTupleType
class DocumentTuple(Tuple):
    """ Document Tuple

    This tuple is the publicly exposed Document

    """
    __tupleType__ = docDbTuplePrefix + 'DocumentTuple'

    #:  The unique key of this document
    key: str = TupleField()

    #:  The unique key of this document
    modelSetKey: str = TupleField()

    #:  The document data
    document: Dict = TupleField()

File: peek_plugin_docdb/plugin-module/DocumentTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {docDbTuplePrefix} from "./_private/PluginNames";
import {DocDbDocumentTypeTuple} from "./DocDbDocumentTypeTuple";


@addTupleType
export class DocumentTuple extends Tuple {
    public static readonly tupleName = docDbTuplePrefix + "DocumentTuple";

    //  The unique key of this document
    key: string;

    // This Document Type ID
    documentType: DocDbDocumentTypeTuple = DocDbDocumentTypeTuple();

    // The document data
    document: {} = {};

    constructor() {
        super(DocumentTuple.tupleName)
    }
}